walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Set<String> attendees = new HashSet<String>(request.getAttendees());
+    List<TimeRange> initalRanges = new ArrayList<TimeRange>();
+    List<TimeRange> results = new ArrayList<TimeRange>();
+    final int minMeetingTime = 15;
+    
+    for(int start = 0; start <= TimeRange.END_OF_DAY; start += minMeetingTime) {
```

Comment:
Adding a comment to explain what you plan to do would help readability. In general, when you are about to do something interesting, it should be preceded by a comment;

// To find a time, the following will first produce a set of candidate blocks and then exclude any blocks currently assigned. Finally, the blocks of time left over will be consolidated.